docs/windows-design.md

```diff
@@ -0,0 +1,261 @@
+container traffic with OpenFlow rules. OpenFlow entries are installed to implement Pod-to-Pod and
+Pod-to-external connectivity. Service traffic is handled by kube-proxy, and a specific
+OpenFlow rule is installed to output these packets to the gateway port.
+
+CNIAdd request might be called multiple times for a given Pod. This is because kubelet on Windows
+assumes CNIAdd is an idempotent event, and it uses this event to query the Pod networking status.
+Antrea needs to identify the container type (sandbox or workload) from the CNIAdd request:
+* we create the HNS Endpoint only when the request is for the sandbox container
+* we attach the HNS Endpoint no matter whether it is a sandbox container or a workload container.
+
+### Gateway port configuration
+
+The gateway port is created during the Antrea Agent initialization phase, and the address of the interface
+should be the first IP in the subnet. The port is an OVS internal port and its default name is "gw0".
+
+The gateway port is used to help implement L3 connectivity for the containers, including Pod-to-external,
+Pod-to-Service, and Node-to-Pod. For Pod-to-external, and Pod-to-Service cases, OpenFlow entries are
+installed in order to output these packets to the host on the gateway port. To ensure the packet is forwarded
+correctly on the host, the IP-Forwarding feature should be enabled on the network adapter of the
+gateway port.
+
+A routing entry for traffic from the Node to the local Pod subnet is needed on the Windows host to ensure 
+that the packet can enter the OVS pipeline on the gateway port. This routing entry is added when the "gw0"
+is created.
+
+Every time a new Node joins the cluster, a host routing entry on the gateway port is required, and the
+remote subnet CIDR should be routed with the remote gateway address as the nexthop.
+
+### Tunnel port configuration
+
+Tunnel port configuration should be similar to Antrea on Linux:
+* tunnel port is created after OVS bridge is added; 
+* a flow-based tunnel with the appropriate remote address is created for each Node in the cluster with OpenFlow.
+
+The only difference with Antrea on Linux is that the tunnel local address is required when creating the tunnel
+port (provided wth `local_ip` option). This local address is the one configured on the OVS bridge. 
+
+### OVS bridge interface configuration
+
+Since OVS is also responsible for taking charge of the network of the host, an interface for the OVS bridge
+is required on which the host network settings are configured. It is created and enabled when creating
+the OVS bridge, and the MAC address should be changed to be the same as the uplink interface. Then the IP 
+address and the route entries originally configured on the uplink interface should also be migrated to
+the interface.
+
+The packet that is sent to/from the Windows host should be forwarded on this interface. So the OVS bridge
```

Comment:
s/The packet that is sent to/from the Windows host/The packets that are sent to/from the Windows host